src/sim-host/ui/sim-host.js

```diff
@@ -1,29 +1,68 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
-var db = require('db'),
+var customElements = require('./custom-elements'),
+    db = require('db'),
+    dialog = require('dialog'),
     Messages = require('messages'),
-    customElements = require('./custom-elements'),
-    socket = require('../protocol/socket'),
-    dialog = require('dialog');
+    Q = require('q'),
+    socket = require('../protocol/socket');
+
+var COLLAPSED_PANELS_KEY = 'collapsed-panels';
 
 var plugins;
 var pluginHandlers = {};
 var serviceToPluginMap = {};
+var initSocketPromise = socket.initialize(pluginHandlers, serviceToPluginMap);
 
-customElements.initialize();
-socket.initialize(pluginHandlers, serviceToPluginMap);
+customElements.initialize(changePanelVisibilityCallback);
 
 window.addEventListener('DOMContentLoaded', function () {
     sizeContent();
-
-    // Initialize standard modules, then plugins
-    db.initialize().then(initializePlugins);
+    Q.all([db.initialize(), initSocketPromise]).then(function () {
+        initializePlugins();
+
+        // Some panels, like geolocation, need to be fully initialized before they can be hidden, otherwise they will
+        // stop working. For that reason, we restore the initial collapse state to the panels only after plugin
+        // initialization.
+        getCollapsedPanels().forEach(function (panelId) {
+            document.getElementById(panelId).cordovaCollapsed = true;
+        });
+    }).done();
 });
 
 window.addEventListener('resize', function () {
     sizeContent();
 });
 
+function changePanelVisibilityCallback(id, isNowCollapsed) {
+    var collapsedPanels = getCollapsedPanels();
+    var index = collapsedPanels.indexOf(id);
+
+    if (isNowCollapsed && index === -1) {
+        collapsedPanels.push(id);
+    } else if (!isNowCollapsed && index > -1) {
+        collapsedPanels.splice(index, 1);
+    }
+
+    db.saveObject(COLLAPSED_PANELS_KEY, collapsedPanels);
+}
+
+function getCollapsedPanels() {
+    var collapsedPanels;
+
+    try {
+        collapsedPanels = db.retrieveObject(COLLAPSED_PANELS_KEY);
+    } catch (err) {
+        // Do nothing, default empty array will be used.
+    }
+
+    if (!collapsedPanels || !Array.isArray(collapsedPanels)) {
```

Comment:
Sounds good! 
I was taking a look at the implementation of [retrieveObject](https://github.com/Microsoft/cordova-simulate/blob/master/src/modules/db.js#L122), and the exception could be produced by JSON.parse, so maybe we can handle the exception in there, and return some default value, null or undefined. To avoid delegating the responsibility of handling the error of JSON.parse to the consumers of this API. But this is not needed to be done as part of this PR, we can think about it fix it later, just in case we miss something.